tools/ci/run_tc.py

```diff
@@ -246,7 +250,65 @@ def set_variables(event):
         os.environ["GITHUB_BRANCH"] = branch
 
 
+def task_url(task_id):
+    root_url = os.environ['TASKCLUSTER_ROOT_URL']
+    if root_url == 'https://taskcluster.net':
+        queue_base = "https://queue.taskcluster.net/v1/task"
+    else:
+        queue_base = root_url + "/api/queue/v1/task"
+
+    return "%s/%s" % (queue_base, task_id)
+
+
+def download_artifacts(artifacts):
+    artifact_list_by_task = {}
+    for artifact in artifacts:
+        base_url = task_url(artifact["task"])
+        if artifact["task"] not in artifact_list_by_task:
+            resp = urlopen(base_url + "/artifacts")
+            artifacts_data = json.load(resp)
+            artifact_list_by_task[artifact["task"]] = artifacts_data
+
+        artifacts_data = artifact_list_by_task[artifact["task"]]
+        print("DEBUG: Got artifacts %s" % artifacts_data)
+        found = False
+        for candidate in artifacts_data["artifacts"]:
+            print("DEBUG: candidate: %s pattern: %s" % (candidate["name"], artifact["artifact"]))
```

Comment:
Nit: I'd prefer using `glob` instead of `pattern` here, saving the trouble of people guessing whether this is a regex.